pkg/utils/config_checker.go

```diff
@@ -0,0 +1,90 @@
+package utils
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/apiserver/pkg/server/healthz"
+)
+
+var _ healthz.HealthChecker = &configChecker{}
+
+// configChecker is used to notify container to restart when config files updated
+type configChecker struct {
+	name        string
+	configfiles []string
+	checksum    [16]byte
+}
+
+// NewConfigChecker
+// name could be any string.
+// configfiles should be the same as your target container are using now.
+//
+// Example:
+// type configCheckerServer struct {
+// 	checkers []heathz.HealthChecker
+// }
+//
+// func NewConfigCheckerServer(checkers []healthz.HealthChecker) *configCheckerServer {
+// 	return &configCheckerServer{checkers: checkers}
+// }
+//
+// func (s *configCheckerServer) ServerHttp(rw http.ResponseWriter, r *http.Request) {
+// 	for _, c := range s.chekers {
+// 		if c.Name() == r.URL {
+// 			if err := c.Check(); err != nil {
+// 				rw.WriteHeader(500)
+// 			} else {
+// 				rw.WriteHeader(200)
+// 			}
+// 		}
+// 	}
+// }
+//
+// Containers in the same pod of configCheckServer can add liveness probe and be triggered restart as config files be modified.
+func NewConfigChecker(name string, configfiles ...string) (*configChecker, error) {
+	checksum, err := load(configfiles)
+	if err != nil {
+		return nil, err
+	}
+	return &configChecker{
+		name:        name,
+		configfiles: configfiles,
+		checksum:    checksum,
+	}, nil
+}
+
+// Name return the name fo the configChecker
+func (c *configChecker) Name() string {
+	return c.name
+}
+
+// Check would return nil if current configfiles's checksum is equal to cached checksum
+// If checksum not equal, it will return err and update cached checksum with current checksum
+// Note that: configChecker performs a instant update after it returns err, so DO NOT use one configChecker for multible containers!!!
+func (cc *configChecker) Check(_ *http.Request) error {
+	newChecksum, err := load(cc.configfiles)
```

Comment:
I think you might want to add a lock around this to ensure it is thread safe

Reply:
Added locker.